astropy/time/core.py

```diff
@@ -803,12 +803,15 @@ def _apply(self, method=None, *args, **kwargs):
             # Apply the method to any value arrays (though skip if there is only
             # a single element and the method would return a view, since in
             # that case nothing would change).
-            if method is not None and val is not None:
-                if method == 'copy' or method == 'flatten' and val.size == 1:
+            if method is not None and val is not None and hasattr(val, method):
+                if val.size > 1 or method == 'copy':
```

Comment:
In principle, we control these internal attributes, so I think it may be best here to keep it as is; in some sense, if it fails here, it would be a good reason to look over the code carefully anyway!